### PR TITLE
Channel archive

### PR DESCRIPTION
**Pull request by thisisjofrank**
### Description

I'm sorry this is so huge

I've added in a chat archive which will retrieve historical chat messages from Azure blob storage.
Added archive storage that can seek in reverse.
- Set up in devrel azure portal (called FFSStorage)
Added a history iterator that will iterate over messages in reverse.
Added side-by-side functions app for ingress.
- Have set up an ably reactor rule for Azure in the FullyFeaturedScalableChat app in the devrel ably account
- Have added the relevant github secrets to the GH 
- Set up in Azure portal (called FFSFunctionsApp)
Fixed up the yaml
Fixed up some linter rules
Created a useArchive hook
Laid foundations for message enhancement/formatting
Fixed broken tests, removed babel, use js-test
Updated readme

### PR Checklist

Please use this checklist to track the completeness of your PR. We're always happy to assist you in completing these steps!

- [x] No linting errors/warnings are introduced.
- [x] Unit tests are all passing.
- [x] The feature/fix has been tested locally.
- [x] The feature/fix satisfies the acceptance criteria described in issue: #.

### CODE REVIEW

File: app/src/components/Channels/ChannelList.jsx [hunk 1]
```diff
@@ -1,5 +1,7 @@
 import React from "react";
 import { Link } from "react-router-dom";
+import { useAuth } from "../../AppProviders";
```
Comment (marcduiker) [inline app/src/components/Channels/ChannelList.jsx:3]: Do we need this here?

File: app/src/components/Chat/Extensions/SwearFilter.jsx [hunk 2]
```diff
@@ -0,0 +1,3 @@
+export default (renderingItem) => {
+    renderingItem.data.text = renderingItem.data.text.replace("crap", "****");
```
Comment (marcduiker) [inline app/src/components/Chat/Extensions/SwearFilter.jsx:2]: This will eventually be replaced with a content moderation service?

**Review by marcduiker — APPROVED** [review]

File: app/src/components/Chat/Extensions/SwearFilter.jsx (hunk 2 above)
Comment (thisisjofrank) [reply app/src/components/Chat/Extensions/SwearFilter.jsx:2]: haha, yeah <3

File: app/src/components/Channels/ChannelList.jsx (hunk 1 above)
Comment (thisisjofrank) [reply app/src/components/Channels/ChannelList.jsx:3]: nope! Removing

**Pull request merged**